File: java-web-performance-webflux/src/main/java/com/binghe/MemberService.java
package com.binghe;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import reactor.core.publisher.Mono;

@Service
@RequiredArgsConstructor
public class MemberService {

    private final MemberRepository memberRepository;

    @Transactional(readOnly = true)
    public Mono<Member> findById(Long id) {
        return memberRepository.findById(id);
    }
}

File: kotlin-grpc-coroutine-spring-boot/grpc-server/src/main/kotlin/com/binghe/member/MemberService.kt
package com.binghe

import kotlinx.coroutines.reactor.awaitSingleOrNull
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class MemberService @Autowired constructor(private val memberRepository: MemberRepository) {

    suspend fun findById(id: Long): Member? {
        return memberRepository.findById(id)
            .awaitSingleOrNull()
    }
}